pkg/controllers/membercluster/v1beta1/membercluster_controller.go

```diff
@@ -489,7 +525,7 @@ func (r *Reconciler) aggregateJoinedCondition(mc *clusterv1beta1.MemberCluster)
 	reportedAgents := make(map[clusterv1beta1.AgentType]bool)
 	for _, agentStatus := range mc.Status.AgentStatus {
 		if !r.agents[agentStatus.Type] {
-			klog.V(2).InfoS("Ignoring unexpected agent type status", "agentStatus", agentStatus)
+			_ = controller.NewUnexpectedBehaviorError(fmt.Errorf("find an unexpected agent type %s for member cluster %s", agentStatus.Type, mc.Name))
```

Comment:
prefer to add a error log here so that it will print out the error location and line number and it's easier for debugging though we'll log the error in the NewUnexpectedBehaviorError func.

Reply:
so we have two back to back error pting here? I was wondering if the functions in the controller package can print out the call stack.

Reply:
looks like this func https://pkg.go.dev/runtime/debug#Stack could help to print the stack trace. But i find that k8 does not use this func a lot.

https://github.com/search?q=org%3Akubernetes+++%22debug.Stack%28%29%22+language%3AGo&type=code

not sure if it's ok for production code, eg, will it consume too much memory.

Reply:
perf shouldn't be a problem as this is supposed to be very rare.